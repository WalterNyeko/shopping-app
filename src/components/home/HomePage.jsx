import React, { Fragment } from 'react';
import UpperNav from '../../containers/commons/UpperNav';
import SideNav from '../../containers/commons/SideNav';

const HomePage = (props) => {
    const { content, departmentId } = props;
    return (
        <Fragment>
            <UpperNav/>
            <SideNav 
                content={content}
                departmentId={departmentId}/>
        </Fragment>
    )
}
export default HomePage;
