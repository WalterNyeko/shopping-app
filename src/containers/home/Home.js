import React, { Component, Fragment } from "react";
import LandingPageComponent from "../../containers/home/LandingPage";

class Home extends Component {
 

  render() {
    return (
      <Fragment>
        <LandingPageComponent />
      </Fragment>
    );
  }
}

export default Home;
