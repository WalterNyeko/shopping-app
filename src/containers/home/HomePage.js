import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import HomePageComponent from "../../components/home/HomePage";
import { getDepartments } from "../../store/actions/Departments";

class HomePage extends Component {
  componentWillMount = () => {
    const { getDepartments } = this.props;
    getDepartments();
  };

  render() {
    const {
      departmentId,
      allDepartments: { departments },
      handleSearch,
      handleInputChange,
      dontRender,
      content,
      categoryId
    } = this.props;
    return (
      <Fragment>
        <HomePageComponent
          content={content}
          dontRender={dontRender}
          categoryId={categoryId}
          departmentId={departmentId}
          departments={departments}
          handleSearch={handleSearch}
          handleInputChange={handleInputChange}
        />
      </Fragment>
    );
  }
}
const mapStateToProps = state => ({
  allDepartments: state.departments
});
export default connect(
  mapStateToProps,
  { getDepartments }
)(HomePage);
