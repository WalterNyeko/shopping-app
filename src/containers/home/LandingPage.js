import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import LandingPageComponent from "../../components/home/LandingPage";
import { getProductsAfterSearch } from "../../store/actions/Products";

export class LandingPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      query: ""
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
  }

  handleInputChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  handleSearch = event => {
    event.preventDefault();
    const { getProductsAfterSearch } = this.props;
    const data = {
      query_string: this.state.query,
      page: 1,
      limit: 8,
      description_length: 100
    };
    getProductsAfterSearch(data);
  };

  render() {
    let theRows;
    if (this.props.allProducts.products) {
      const { rows } = this.props.allProducts.products;
      theRows = rows;
    } else {
      theRows = { count: 0, rows: [] };
    }
    return (
      <Fragment>
        <LandingPageComponent
          handleSearch={this.handleSearch}
          handleInputChange={this.handleInputChange}
          products={theRows}
          afterSearch={this.state.query}
          showSearchField
        />
      </Fragment>
    );
  }
}
const mapStateToProps = state => ({
  allProducts: state.products
});
export default connect(
  mapStateToProps,
  { getProductsAfterSearch }
)(LandingPage);
