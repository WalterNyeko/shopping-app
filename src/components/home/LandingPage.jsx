import React, { Fragment } from "react";
import HomePage from "../../containers/home/HomePage";
import AllProducts from "../../containers/products/AllProducts";

const LandingPage = props => {
  const { products, handleInputChange, handleSearch, afterSearch } = props;
  return (
    <Fragment>
      <HomePage
        content={<AllProducts products={products} afterSearch={afterSearch} />}
        handleInputChange={handleInputChange}
        handleSearch={handleSearch}
      />
    </Fragment>
  );
};
export default LandingPage;
