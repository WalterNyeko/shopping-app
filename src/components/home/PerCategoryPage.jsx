import React, { Fragment } from 'react';
import HomePage from '../../containers/home/HomePage';
import CategoryProducts from '../../containers/products/CategoryProducts';

const PerCategoryPage = (props) => {
    const { categoryId } = props;
    return (
        <Fragment>
            <HomePage 
                content={<CategoryProducts categoryId={categoryId}/>}/>
        </Fragment>
    )
}
export default PerCategoryPage;