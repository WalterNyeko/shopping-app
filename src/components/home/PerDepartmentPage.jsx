import React, { Fragment } from "react";
import HomePage from "../../containers/home/HomePage";
import ProductsPerDepartment from "../../containers/products/DepartmentProducts";

const PerDepartmentPage = props => {
  const { departmentId } = props;
  return (
    <Fragment>
      <HomePage
        content={<ProductsPerDepartment departmentId={departmentId} />}
        departmentId={departmentId}
      />
    </Fragment>
  );
};
export default PerDepartmentPage;
