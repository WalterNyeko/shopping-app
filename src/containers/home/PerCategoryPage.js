import React, { Component, Fragment } from 'react';
import PerCategoryPage from '../../components/home/PerCategoryPage';

class LandingPage extends Component {
    render() {
        const { categoryId } = this.props.match.params;
        return (
            <Fragment>
                <PerCategoryPage categoryId={categoryId}/>
            </Fragment>
        )
    }
}
export default LandingPage;
