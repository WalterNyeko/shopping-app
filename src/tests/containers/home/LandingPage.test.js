import React from 'react';
import { shallow } from 'enzyme';
import { LandingPage } from '../../../containers/home/LandingPage';


describe('LandingPage', () => {
    const props = {
        allProducts: {
            products: {count: 11, rows: []}
        },
        getProductsAfterSearch: jest.fn(),
    }
    const wrapper = shallow(<LandingPage {...props}/>);
    it('should render without crashing', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call handleSearch when the function is called', () => {
        const spy = jest.spyOn(wrapper.instance(), "handleSearch");
        const event = {target: {name: '',value: ''}, preventDefault: jest.fn()}
        wrapper.instance().handleSearch(event);
        expect(spy).toHaveBeenCalled();
    });

    it('should render well when there is no products', () => {
        wrapper.setProps({allProducts: {}});
        expect(wrapper).toMatchSnapshot();
    });

    it("should update the state with the new value whenever input changes on search field", () => {
        const spy = jest.spyOn(wrapper.instance(), 'handleInputChange');
        const event = { target: { name: 'query', value: 'Italia'}}
        wrapper.instance().handleInputChange(event);
        expect(spy).toHaveBeenCalled();
        expect(wrapper.instance().state.query).toEqual("Italia");
      });
})