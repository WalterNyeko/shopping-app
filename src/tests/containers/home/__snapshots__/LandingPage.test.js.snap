// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should render well when there is no products 1`] = `ShallowWrapper {}`;

exports[`LandingPage should render without crashing 1`] = `ShallowWrapper {}`;
